session_test.go

```diff
@@ -273,6 +273,68 @@ func TestAccept(t *testing.T) {
 	}
 }
 
+func TestClose_closeTimeout(t *testing.T) {
+	conf := testConf()
+	conf.StreamCloseTimeout = 10 * time.Millisecond
+	client, server := testClientServerConfig(conf)
+	defer client.Close()
+	defer server.Close()
+
+	if client.NumStreams() != 0 {
+		t.Fatalf("bad")
+	}
+	if server.NumStreams() != 0 {
+		t.Fatalf("bad")
+	}
+
+	wg := &sync.WaitGroup{}
+	wg.Add(2)
+
+	// Open a stream on the client but only close it on the server.
+	// We want to see if the stream ever gets cleaned up on the client.
+
+	go func() {
+		defer wg.Done()
+		_, err := client.OpenStream()
```

Comment:
Hm, do Yamux streams close-via-finalizer? If so it seems possible for this test to pass even without your code change as the GC would be implicitly the client-side of the stream.

If this test fails without your patch I think that indicates both this test and the patch are working as intended.

Reply:
Although another way to structure this test might be for the clientside to write in a loop and assert the first error enountered is a `connection reset by peer` and that it occurs after the stream close timeout.

Relying _more_ on timings is rarely a way to make a test more resilient though, so perhaps the added complexity isn't worth it.

Reply:
We don't use any finalizers and nothing else will clean a stream up except a session close (which happens way later here). I can be extra sure for the test by retaining a reference though if that helps. :)

Also I did verify the test fails before, I usually do TDD for bugs like this so I had the test written out first.

Reply:
Kept it in scope and also added a write fail test.